src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -111,20 +111,36 @@ public class TimetableSnapshot {
    * more than once.
    * TODO RT_AB: More general handling of all realtime indexes outside primary data structures.
    */
-  private SetMultimap<StopLocation, TripPattern> patternsForStop = HashMultimap.create();
+  private final SetMultimap<StopLocation, TripPattern> patternsForStop;
```

Comment:
@vpaturet didn't you say earlier that this field was the only one that was not final? Looks like it is indeed final, so this class is essentially a record. Not that I'm advocating changing it to one, just pointing out this detail.

Reply:
I was referring to this field:
`private boolean dirty`
which needs to be updated during the lifecycle of the buffer snapshot, but is not used in the published snapshot.

Reply:
Sorry, I meant to put the comment on `private final boolean readOnly;` but your comment still applies: the non-final field is the one right below it (`dirty`) which I for some reason didn't notice.